source/Ship.cpp

```diff
@@ -752,6 +752,22 @@ int64_t Ship::ChassisCost() const
 
 
 
+// Get the maximum firing energy among all installed weapons that
+// don't fire continuously (like beams do).
+double Ship::MaxFiringEnergy() const
+{
+	double maxFiringEnergy = 0.0;
+	for(const auto &it : outfits)
+	{
+		if (it.first->IsWeapon() && it.first->Reload() > 1) {
```

Comment:
This shouldn't care whether or not the energy drain is continuous or not - nothing is stopping a beam weapon from having a massive energy drain, after all.

Reply:
Just tried a sunbeam on a firebird with only the smallest fuel cell installed, and I can confirm that it didn't fire.

I think a solution to this problem might be more complicated than just including continuous weapons in the max calculation. On the same firebird, a beam laser fires just fine even though it has a firing energy of 0.5 in weapons.txt, indicating that as long as a ship has enough `energy generation / second`, it will be able to fire a weapon. On the other hand, an energy blaster doesn't fire at all, even though its `firing energy / second` and `firing energy / shot` are well below the ship's `energy generation / second`. 

As far as I can tell, continuous weapons can fire as long as the ship has enough energy per second, but packeted weapons require a battery of size at least `firing energy / shot` to fire. I'll update the code in a few hours.

Reply:
The key metric is per frame, not per second. All weapons consume their total firing energy each time they fire, and if that amount is available at the time of weapons checking, the weapon can fire--even if there is no energy storage and even if the weapon has a reload > 1. 

So if a ship has an energy generation of >= 5.2 per frame (>= 312 per second), it can possibly fire a Sunbeam (firing energy = 5.2 per shot) even without any energy storage. (If things like movement, shield repair, or other weapons have consumed energy this frame, then the ship may no longer have the needed energy in the current frame, in which case it would not be able to fire the Sunbeam.)

Reply:
That makes sense. So the energy generation listed in the outfitter is `energy / second` but what really matters for firing weapons is `energy / frame`.  https://github.com/endless-sky/endless-sky/pull/4177/commits/a8a942f4a948fb85b07a8b86ddfb8c1ad3f30a9e takes this into account, checking all weapons (including those that have a reload of 1) against the total available `energy / frame` of the ship, which I think should be `solar collection + energy generation + battery`.